apps/web/src/hooks/useRegisterNameCallback.ts

```diff
@@ -22,10 +27,46 @@ export function useRegisterNameCallback(
   value: bigint | undefined,
   years: number,
   discountKey?: `0x${string}`,
-  validationData?: `0x${string}`,
-) {
-  const { address } = useAccount();
+  validationData?: `0x${string}`,) {
+  if (discountKey) {
+    console.log('discountKey', discountKey);
+    console.log('validationData', validationData);
+  }
+  const { address, chainId } = useAccount();
+  
+  const account = useAccount();
+  const { data: client } = useWalletClient();
+  const { writeContract } = useWriteContract();
+  const network = chainId === baseSepolia.id ? chainId : base.id;
+  const { writeContractsAsync } = useWriteContracts(); // For Smart Contract Wallet calls
+  
+  
+  const {data: availableCapacities} = useCapabilities({
+      account: address
+    });
+  
+  const capabilities = useMemo(() => {
+    if (!account.isConnected || !chainId || !availableCapacities) {
+      return {};
+    }
+    const chainCapabilities = availableCapacities[chainId];
+    if (chainCapabilities["paymasterService"] && chainCapabilities["paymasterService"].supported) {
+      return {
+        paymasterService: {
+          // url: `${document.location.origin}/api/paymaster`
+          url: 'https://api.developer.coinbase.com/rpc/v1/base-sepolia/1IhTcPOmhK5aEq-4WqRZMJoOh0oPenD2'
+        }
+      }
+    }
+    return {};
+  }, [availableCapacities, chainId]);  
 
+  if (chainId === undefined) {
+    console.error(
+      'useRegisterNameCallback: Unable to create name registration callback; chainId must be defined.',
+    );
+    return () => {};
+  }
```

Comment:
the early return here will cause client side crashes if the code ever runs because this hook will be running a variable number of hooks